sda-commons-dependencies/build.gradle

```diff
@@ -134,9 +134,12 @@ dependencies {
     api "net.javacrumbs.json-unit:json-unit-assertj:2.18.1"
 
     // sda-commons-server-testing
-    api "junit:junit:4.13"
-    api "org.assertj:assertj-core:3.16.1"
-    api "org.mockito:mockito-core:3.3.3"
+    api "org.hamcrest:hamcrest:2.2"
+    // hamcrest-core is deprecated and overlaps with hamcrest, in hamcrest-core 2.2 is only one
+    // class named HamcrestCoreIsDeprecated. We force this version to avoid complex exclude and
+    // dependency substitution configuration as there is no conflict with hamcrest in the code base.
+    // Please do not upgrade hamcrest-core.
+    api "org.hamcrest:hamcrest-core:2.2"
```

Comment:
I used Gradle's "because" syntax to explain certain versions and like that pattern.